Enable swift-markdown again

I would like to revert 1566d6b2be2eac7a0f5d52a01321520b5415dbef, but compilation errors occur on swift-cmark.

```console
$ swift build --product swift-format --experimental-swift-sdk wasm32-unknown-wasi
...
/swift-format/.build/checkouts/swift-cmark/extensions/core-extensions.c:22:1: error: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
   22 | CMARK_DEFINE_ONCE(registered);
      | ^
      | int
/swift-format/.build/checkouts/swift-cmark/extensions/core-extensions.c:22:19: error: a parameter list without types is only allowed in a function definition
   22 | CMARK_DEFINE_ONCE(registered);
      |                   ^
/swift-format/.build/checkouts/swift-cmark/extensions/core-extensions.c:30:3: error: call to undeclared function 'CMARK_RUN_ONCE'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
   30 |   CMARK_RUN_ONCE(registered, register_plugins);
      |   ^
/swift-format/.build/checkouts/swift-cmark/extensions/core-extensions.c:30:18: error: use of undeclared identifier 'registered'
   30 |   CMARK_RUN_ONCE(registered, register_plugins);
      |                  ^
4 errors generated.
```

```console
$ swift build --product swift-format --experimental-swift-sdk wasm32-unknown-wasi -Xcc -Wno-error=implicit-int -Xcc -Wno-error=implicit-function-declaration
...
/swift-format/.build/checkouts/swift-cmark/extensions/core-extensions.c:22:1: warning: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
   22 | CMARK_DEFINE_ONCE(registered);
      | ^
      | int
/swift-format/.build/checkouts/swift-cmark/extensions/core-extensions.c:22:19: error: a parameter list without types is only allowed in a function definition   22 | CMARK_DEFINE_ONCE(registered);
      |                   ^
/swift-format/.build/checkouts/swift-cmark/extensions/core-extensions.c:30:3: warning: call to undeclared function 'CMARK_RUN_ONCE'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
   30 |   CMARK_RUN_ONCE(registered, register_plugins);
      |   ^
/swift-format/.build/checkouts/swift-cmark/extensions/core-extensions.c:30:18: error: use of undeclared identifier 'registered'
   30 |   CMARK_RUN_ONCE(registered, register_plugins);
      |                  ^
2 warnings and 2 errors generated.
```

```
/swift-format/.build/checkouts/swift-cmark/extensions/core-extensions.c:22:19: error: a parameter list without types is only allowed in a function definition
   22 | CMARK_DEFINE_ONCE(registered);
      |                   ^
```

`CMARK_DEFINE_ONCE` is a macro, so this shouldn't be an error.

https://github.com/apple/swift-cmark/blob/2c47322cb32cbed555f13bf5cbfaa488cc30a785/src/include/mutex.h#L70

--- a/Package.swift
+++ b/Package.swift
@@ -166,7 +166,7 @@ var dependencies: [Package.Dependency] {
   } else {
     return [
       .package(url: "https://github.com/apple/swift-argument-parser.git", from: "1.2.2"),
-      .package(url: "https://github.com/apple/swift-markdown.git", from: "0.2.0"),
+      .package(url: "https://github.com/kkk669/swift-markdown.git", branch: "wasm32-wasi"),
       .package(url: "https://github.com/apple/swift-syntax.git", branch: "main"),
     ]
   }

--- a/Package.swift
+++ b/Package.swift
@@ -47,7 +47,7 @@ let package = Package(
     .target(
       name: "SwiftFormat",
       dependencies: [
-        .product(name: "Markdown", package: "swift-markdown", condition: .when(platforms: [.macOS, .iOS, .linux, .windows])),
+        .product(name: "Markdown", package: "swift-markdown"),
         .product(name: "SwiftSyntax", package: "swift-syntax"),
         .product(name: "SwiftSyntaxBuilder", package: "swift-syntax"),
         .product(name: "SwiftOperators", package: "swift-syntax"),

--- a/Sources/SwiftFormat/Core/DocumentationComment.swift
+++ b/Sources/SwiftFormat/Core/DocumentationComment.swift
@@ -10,8 +10,6 @@
 //
 //===----------------------------------------------------------------------===//
 
-#if !os(WASI)
-
 import Markdown
 import SwiftSyntax
 
@@ -345,5 +343,3 @@ private struct SimpleFieldMarkupRewriter: MarkupRewriter {
     return Text(String(nameAndRemainder[1]))
   }
 }
-
-#endif

--- a/Sources/SwiftFormat/Core/Pipelines+Generated.swift
+++ b/Sources/SwiftFormat/Core/Pipelines+Generated.swift
@@ -239,9 +239,7 @@ class LintPipeline: SyntaxVisitor {
     visitIfEnabled(NoLeadingUnderscores.visit, for: node)
     visitIfEnabled(OmitExplicitReturns.visit, for: node)
     visitIfEnabled(UseTripleSlashForDocumentationComments.visit, for: node)
-#if !os(WASI)
     visitIfEnabled(ValidateDocumentationComments.visit, for: node)
-#endif
     return .visitChildren
   }
   override func visitPost(_ node: FunctionDeclSyntax) {
@@ -251,9 +249,7 @@ class LintPipeline: SyntaxVisitor {
     onVisitPost(rule: NoLeadingUnderscores.self, for: node)
     onVisitPost(rule: OmitExplicitReturns.self, for: node)
     onVisitPost(rule: UseTripleSlashForDocumentationComments.self, for: node)
-#if !os(WASI)
     onVisitPost(rule: ValidateDocumentationComments.self, for: node)
-#endif
   }
 
   override func visit(_ node: FunctionParameterSyntax) -> SyntaxVisitorContinueKind {
@@ -344,18 +340,14 @@ class LintPipeline: SyntaxVisitor {
     visitIfEnabled(AllPublicDeclarationsHaveDocumentation.visit, for: node)
     visitIfEnabled(BeginDocumentationCommentWithOneLineSummary.visit, for: node)
     visitIfEnabled(UseTripleSlashForDocumentationComments.visit, for: node)
-#if !os(WASI)
     visitIfEnabled(ValidateDocumentationComments.visit, for: node)
-#endif
     return .visitChildren
   }
   override func visitPost(_ node: InitializerDeclSyntax) {
     onVisitPost(rule: AllPublicDeclarationsHaveDocumentation.self, for: node)
     onVisitPost(rule: BeginDocumentationCommentWithOneLineSummary.self, for: node)
     onVisitPost(rule: UseTripleSlashForDocumentationComments.self, for: node)
-#if !os(WASI)
     onVisitPost(rule: ValidateDocumentationComments.self, for: node)
-#endif
   }
 
   override func visit(_ node: IntegerLiteralExprSyntax) -> SyntaxVisitorContinueKind {

--- a/Sources/SwiftFormat/Rules/BeginDocumentationCommentWithOneLineSummary.swift
+++ b/Sources/SwiftFormat/Rules/BeginDocumentationCommentWithOneLineSummary.swift
@@ -90,7 +90,6 @@ public final class BeginDocumentationCommentWithOneLineSummary: SyntaxLintRule {
 
   /// Diagnose documentation comments that don't start with one sentence summary.
   private func diagnoseDocComments(in decl: DeclSyntax) {
-#if !os(WASI)
     guard
       let docComment = DocumentationComment(extractedFrom: decl),
       let briefSummary = docComment.briefSummary
@@ -108,7 +107,6 @@ public final class BeginDocumentationCommentWithOneLineSummary: SyntaxLintRule {
         diagnose(.terminateSentenceWithPeriod(trailingText), on: decl)
       }
     }
-#endif
   }
 
   /// Returns all the sentences in the given text.

--- a/Sources/SwiftFormat/Rules/ValidateDocumentationComments.swift
+++ b/Sources/SwiftFormat/Rules/ValidateDocumentationComments.swift
@@ -10,8 +10,6 @@
 //
 //===----------------------------------------------------------------------===//
 
-#if !os(WASI)
-
 import Foundation
 import Markdown
 import SwiftSyntax
@@ -198,10 +196,3 @@ extension Finding.Message {
     "add a 'Throws:' section to document the errors thrown by '\(funcName)'"
   }
 }
-
-#else
-
-@_spi(Rules)
-public final class ValidateDocumentationComments {}
-
-#endif

